CONTRIBUTING.md

```diff
@@ -71,6 +71,54 @@ Be sure to include any related GitHub issue references in the commit message.  S
 [GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues
 and commits.
 
+### Formatting Code
+
+ We do use Prettier as a tool to ensure a consistent format. A format check step runs as part of our continuous integration and pull requests get failed checks if they don't adhere to the conventions. To make sure your contribution is properly formatted you can run the following:
+
+ ``` shell
+ npm run format-check
+ ```
+
+ The `format` script can also be used to make prettier automatically apply the format guidelines:
+
+ ``` shell
+ npm run format
+ ```
+
+ ### Static Analysis
+
+ As part of our continuous integration workflow we do pass ESlint to all pull requests. To make sure that your contribution passes all the static checks you can use:
+
+ ``` shell
+ npm run lint
+ ```
+
+ ### Tests
+
+ We value highly-tested software. All pull requests should come accompanied by corresponding unit tests. Integration tests are also welcomed. To make sure that your contribution isn't causing any regression we run the test suite as part of our continuous integration process. You should also make sure that all the tests pass before sending your pull request:
+
+ ``` shell
+ npm run test
+ ```
+
+ ## Release Process
+
+ All stable code is hosted at the `main` branch. Release and version management is fully automated and based in commit messages that follow a subset of the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec. Based on the commit message, our continuous integration pipeline will automatically build, package and version bump and tag the codebase. 
+
+ Versioning follows the [semver](https://semver.org) spec and the appropriate patch, minor or major version is bumped depending on the commit messages. Here is the convention used for version bumping based on the commit messages:
```

Comment:
We started to use `X.Y`, just major and minor. Do you think we need `X.Y.Z` for this artifact?

Reply:
Yes. For this library it makes sense. Otherwise, we would have to implement our own automated version bump logic which is not worth it. Also, I presume that if we really want to, and as the GitHub Marketplace promotes process is manual, we can use X.Y ( manual tag and promote ) for releases into the marketplace.

Reply:
If I'm understanding correctly, we choose `X.Y.Z` because there is an GH action that it makes easier the version management. But I don't see any technical reason / need to use it. `X.Y` would be enough.

Reply:
As per the library, there is a technical need to do handle the bumping which implies changing package.json and package-lock.json automatically with every commit. 

On versioning, I did test tuning the library options trying to use X.Y and it does not work. And nothing will work really because it happens that npm follows semver so it needs major.minor.patch. Trying to use a version like 0.1 [will fail](https://stackoverflow.com/questions/16887993/npm-why-is-a-version-0-1-invalid) with something like:

```
✖  fatal     npm ERR! Invalid version: 0.1
```

So either 0.1.0 with no patch bumps and bump on minor or keep the approach that I have now which bumps patch on each commit. I think the latter makes more sense.